### ISSUE TITLE
Is it possible to 

### ISSUE THREAD

**Issue by jkitchin** [abo-abo/swiper#2892]
Is it possible to make a selection buffer that looks like this in ivy? I am familiar with using a transformer for annotations, I was wondering more about the separators.

![image](https://user-images.githubusercontent.com/1610070/126343201-1dd3a423-4b3c-4f70-b091-cf67022e54e0.png)

**Comment by basil-conto** [comment]
> Is it possible to make a selection buffer that looks like this in ivy?

In theory, of course - perhaps even today, by making some clever (read: kludge) usage of display properties or similar.  But it's not a currently supported interface AFAIK.

I guess the correct way to go about this would be to support the `group-function` completion metadata property that's new in Emacs 28. @minad (author of `group-function` and the better integrated and more modular completion packages Vertico, Marginalia, etc.) will know more about this than I.

For some other similar UI discussions see also #2780 and #2847.

**Comment by bdarcus** [comment]
> I guess the correct way to go about this would be to support the `group-function` completion metadata property that's new in Emacs 28.

Yes, this is definitely the right path. I thought Ivy might already have support for `group-function`, but if not, definitely would be great to have.
